src/tv2-common/onTimelineGenerate.ts

```diff
@@ -54,12 +53,19 @@ export interface PieceMetaData {
 		isMix?: boolean
 		isJingle?: boolean
 	}
-	stickySisyfosLevels?: { [key: string]: { value: number; followsPrevious: boolean } }
+	sisyfosPersistMetaData?: SisyfosPersistMetaData
 	mediaPlayerSessions?: string[]
 	mediaPlayerOptional?: boolean
 	modifiedByAction?: boolean
 }
 
+export interface SisyfosPersistMetaData {
+	sisyfosLayers: string[]
+	wantsToPersistAudio?: boolean
+	acceptPersistAudio?: boolean
+	previousPersistMetaDataForCurrentPiece?: SisyfosPersistMetaData
```

Comment:
What is the reason to gather all those levels of previous metadata? 
![Screenshot 2022-03-15 at 17 20 28](https://user-images.githubusercontent.com/9103996/158425038-e1a99c3b-d841-4dab-a171-d251cb6343da.png)

Reply:
It's because of how the executeAction 'cutToCamera' works.
Instead of adding another piece for the camera we cut to, that executeAction finds the current camera piece and then updates that piece with the new camera information (Code is in executeAction.executeActionCutToCamera() around line 1103).
This means that if we adLib to CAM 1, then CAM 2 there is now no information about CAM 1 ever being in the Part.

Lets say we have LIVE 1, LIVE 2, CAM 2 and CAM 3 that all accept persistence and CAM 1 that does not accept persistence (all Pieces wants to persist). If we goes from LIVE 1 to CAM 1, to CAM 2, to LIVE 2, then the expected outcome of the layers to persist should be LIVE 2. If we don't gather the previous metaData the outcome would be that both LIVE 1 and LIVE 2 is persisted because when calculating what layers to persist we don't have any information about CAM 1 ever being in the Part.

So if we cut LIVE 1, CAM 1, CAM 2, CAM 1, CAM 3, CAM 2 instead of creating a Part with the Pieces: [LIVE 1, CAM 1, CAM 2, CAM 1, CAM 3, CAM 2] Sofie creates a Part with the Pieces: [LIVE 1, CAM 2], but we can't ignore the metatData information on all the camera pieces that Sofie disregarded so we have to gather them somewhere.
(Not that we currently only gather previous metaData on executeAction cutToCamera)

Reply:
If you cut: LIVE 1, CAM 2, CAM 3, CAM 2, CAM 3, **CAM 1**, CAM 2, CAM 3, CAM 2, CAM 3
do you still need to know that all of these were taken in this exact order? Can't this list be reset when CAM 1 is taken? And later when CAM 2 and CAM 3 are taken multiple times consecutively, can't you just store [CAM 1, CAM 2, CAM 3]? Does it change anything regarding persistence that they were taken multiple times?